addons/l10n_th/views/report_invoice.xml

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="report_invoice_document" inherit_id="account.report_invoice_document">
+        <xpath expr="//div[@name='address_not_same_as_shipping']//span[@t-field='o.partner_id.vat']" position="after">
+            <t t-if="o.company_id.account_fiscal_country_id.code == 'TH' and not commercial_invoice">
+                <span t-esc="o.partner_id.l10n_th_branch_name"/>
+            </t>
+        </xpath>
+
+        <xpath expr="//div[@name='address_same_as_shipping']//span[@t-field='o.partner_id.vat']" position="after">
+            <t t-if="o.company_id.account_fiscal_country_id.code == 'TH' and not commercial_invoice" >
+                <span t-esc="o.partner_id.l10n_th_branch_name"/>
+            </t>
+        </xpath>
+        <xpath expr="//div[@name='no_shipping']//span[@t-field='o.partner_id.vat']" position="after">
+            <t t-if="o.company_id.account_fiscal_country_id.code == 'TH' and not commercial_invoice" >
+                <span t-esc="o.partner_id.l10n_th_branch_name"/>
+            </t>
+        </xpath>
+        <xpath expr="//h2/span[contains(@t-if, 'posted')]" position="replace">
+            <t t-if="o.move_type == 'out_invoice' and o.state == 'posted'">
+                <span t-if="o.company_id.account_fiscal_country_id.code == 'TH' and not commercial_invoice">Tax Invoice</span>
+                <span t-else="">Invoice</span>
+            </t>
+        </xpath>
```

Comment:
Is it not possible to manage this in the translations?
`position="replace"` should be avoided as much as possible

Reply:
How about we do it like done in other l10n, inherit the report with a `primary="True"` and then override `_get_name_invoice_report` to load the new report when printing?

It would still use replace but it wouldn't impact the existing template => no problems
I'm not a fan of using translations to change the nature of a term completely